apps/s_client.c

```diff
@@ -2172,6 +2172,16 @@ int s_client_main(int argc, char **argv)
              * HTTP/d.d ddd Reason text\r\n
              */
             mbuf_len = BIO_gets(fbio, mbuf, BUFSIZZ);
+            if (mbuf_len < 12) {
+                /* '12' stands for the length of 'HTTP/d.d ddd' */
```

Comment:
Even if compiler doesn't replace strlen("string") with constant (not all compilers do), it's not in performance-critical loop. If you are worried about it executing strlen every time you invoke s_client, then you would have to think twice, three times, four, ... hundred before invoking s_client to start with. I mean can imagine how expensive just starting program is? Or establishing TCP connection? I mean in comparison to one strlen?